test/e2e/kubectl.go

```diff
@@ -72,7 +73,30 @@ const (
 	simplePodPort            = 80
 )
 
-var proxyRegexp = regexp.MustCompile("Starting to serve on 127.0.0.1:([0-9]+)")
+var (
+	proxyRegexp = regexp.MustCompile("Starting to serve on 127.0.0.1:([0-9]+)")
+
+	// Jobs were introduced in v1.1.0, so we don't expect tests that rely
+	// on jobs to work on clusters before that.
+	//
+	// TODO(ihmccreery): remove once we don't care about v1.0 anymore, (tentatively
+	// in v1.3).
+	jobsVersion = version.MustParse("v1.1.0")
+
+	// NodePorts were introduced in v1.1.0, so we don't expect tests that rely
```

Comment:
This is misleading. NodePorts always existed. They were made optional in v1.1.0. The errors you're guarding against are validation errors when clients omit them and servers expect them.